doc/decisions/DR018-Result-pager-design.md

```diff
@@ -0,0 +1,150 @@
+  solution partly as it facilitates future extensions to data set
+  navigation with minimal disruption to the rest of the API.
+
+- By preference, OpenAssetIO opts for delegation over generalization -
+  to try and avoid the "lowest common denominator" trap awaiting
+  anyone who attempts to generalize the interactions between complex
+  systems.
+
+- Delegation also aims to encourage consistency across tools as
+  interactions with the manager are then presented via its native UI.
+
+## Options considered
+
+### Option 1 - "Simple queue"
+
+The pager is a marshal manning a simple queue. There is a large a number
+of results, and they are supplied to the caller a page at a time. Once
+`next` has been called, there is no way to retrieve the data
+for that page again.
+
+Its interface could look like this:
+
+```python
+class Pager:
+
+    def next() -> [Value]
+        """
+        Retrieves the next page of results.
+        """
+
+    def hasMore() -> Bool
+        """
+        Determines if calls to `next` will yield more results.
+        """
+```
+
+#### Pros
+
+- Simple, well understood data access pattern. Implementations are
+  guaranteed that the first page is always accessed.
+- Smaller surface are makes it easier to support across a variety of
+  back end implementations.
+- Discourages direct use of the API for complex workflows, encouraging
+  delegated UX, which can result in a more consistent user experience.
+
+#### Cons
+
+- Accessing data from anything other than the first page requires
+  traversal of all preceding pages.
+- Not all runtime environments support delegated UX and/or have critical
+  mass such that a manager would provide them with suitable UI
+  components.
+- Limiting to a simple queue potentially offsets some argued benefits of
+  a pager object in terms of future extension, and could even be said to
+  be more concisely expressed as an additional token supplied to the
+  main `ManagerInterface` method.
+
+### Option 2 - "Moving window"
+
+The pager is a moving window into the results, facilitating more
+general navigation of the data set.
+
+Its interface could look like this:
+
+```python
+class Pager:
+
+    def get() -> [Value]
+        """
+        Retrieves the results for the current page.
+        """
+
+    def next() -> None
+        """
+        Advances the pager to the next page.
+        """
+
+    def previous() -> None
+        """
+        Rewinds the pager to the previous page.
+        """
+
+    def hasNext() -> Bool
+        """
+        Determines if there are results after the currrent page.
+        """
+
+    def hasPrevious() -> Bool
+        """
+        Determines if there are results before the currrent page.
+        """
+```
+
+#### Pros
+
+- Facilitates more complex uses of the underlying data beyond gated
+  consumption of a fixed stream.
+- Allows runtimes without delegated UX support to build more advanced
+  workflows.
```

Comment:
[Consider]
```suggestion
- Allows runtimes without delegated UX support to build more advanced
  workflows.
- Extendable to more complex traversal concepts such as jumps and
  skips should the need for them arise.
```

Reply:
added [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/985/commits/f91090f12ec180e79114258d32acbc31622a98ef)

Reply:
🤔 Sorry - was sure I added a comment to that effect!

Reply:
Oh, its as I accepted your suggestion so it auto closes the thread.